src/v2/ion/uiSchemaTransformer.js

```diff
@@ -57,8 +57,17 @@ const createUISchema = (remediationValue = []) => {
 const insertUISchema = (transformedResp) => {
   if (transformedResp.currentState) {
     const remediation = transformedResp.currentState.remediation || [];
-
     transformedResp.currentState.remediation = remediation.map(obj => {
+      // For casee where field itself is a form, it has a formname and we are appending the formname to each field
+      // This is so that while making the request we can bundle these key:value pairs under the same key name
+      // We are assuming that when field itself is a form its only one level deep
+      if (obj.value[0] && obj.value[0].name && obj.value[0].form) {
```

Comment:
worth unit test this logic

Reply:
- feels like the logic shall go into `createUISchema` (before the `remediationValue.map`) so that keep `remediation.value` unchanged.
- consequently, we need to change `BaseModel.create` to use `remediation.uiSchema` instead of `remediation.value`